src/home/NotificationCenter.test.tsx

```diff
@@ -940,4 +949,109 @@ describe('NotificationCenter', () => {
+        </Provider>
+      )
+      expect(getByText('CleverTap Message Header')).toBeDefined()
+      expect(getByText('CleverTap Message Body Text')).toBeDefined()
+    })
+
+    it('emits correct events when CTA is pressed', () => {
+      const store = createMockStore({
+        ...storeDataNotificationsDisabled,
+        home: {
+          cleverTapInboxMessages: [mockExpectedCleverTapInboxMessage],
+        },
+      })
+      const { getByText } = render(
+        <Provider store={store}>
+          <NotificationCenter {...getMockStackScreenProps(Screens.NotificationCenter)} />
+        </Provider>
+      )
+
+      fireEvent.press(getByText('CleverTap Message CTA'))
+      expect(store.getActions()).toEqual([
+        fetchAvailableRewards(),
+        openUrl('https://valoraapp.com', false, true),
+      ])
+
+      expect(CleverTap.pushInboxNotificationClickedEventForId).toBeCalledWith(
+        mockExpectedCleverTapInboxMessage.id
+      )
+
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(HomeEvents.notification_select, {
+        notificationType: NotificationType.clevertap_notification,
+        selectedAction: NotificationBannerCTATypes.accept,
+        notificationId: `${NotificationType.clevertap_notification}/${mockExpectedCleverTapInboxMessage.id}`,
+        notificationPositionInList: 0,
+      })
+    })
+
+    it('emits correct events when notification is dismissed', () => {
+      const store = createMockStore({
+        ...storeDataNotificationsDisabled,
+        home: {
+          cleverTapInboxMessages: [mockExpectedCleverTapInboxMessage],
+        },
+      })
+      const { getByText } = render(
+        <Provider store={store}>
+          <NotificationCenter {...getMockStackScreenProps(Screens.NotificationCenter)} />
+        </Provider>
+      )
+
+      fireEvent.press(getByText('dismiss'))
+
+      expect(CleverTap.deleteInboxMessageForId).toBeCalledWith(mockExpectedCleverTapInboxMessage.id)
+
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(HomeEvents.notification_select, {
+        notificationType: NotificationType.clevertap_notification,
+        selectedAction: NotificationBannerCTATypes.decline,
+        notificationId: `${NotificationType.clevertap_notification}/${mockExpectedCleverTapInboxMessage.id}`,
+        notificationPositionInList: 0,
+      })
+    })
+
+    it('emits correct events when notification is displayed', async () => {
+      const store = createMockStore({
+        ...storeDataNotificationsDisabled,
+        home: {
+          cleverTapInboxMessages: [mockExpectedCleverTapInboxMessage],
+        },
+      })
+      const screen = render(
+        <Provider store={store}>
+          <NotificationCenter {...getMockStackScreenProps(Screens.NotificationCenter)} />
+        </Provider>
+      )
+
+      layoutNotificationList(screen)
+
+      await waitFor(() => expect(ValoraAnalytics.track).toHaveBeenCalledTimes(2))
```

Comment:
why 2 times? perhaps we can do the `waitFor` on the assertion for notification_impression instead?